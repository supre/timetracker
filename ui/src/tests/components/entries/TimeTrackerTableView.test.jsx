import React from "react";
import renderer from "react-test-renderer";
import { cleanup, fireEvent, render, screen } from "@testing-library/react";
import View from "./../../../components/entries/TimeTrackerTableView";

const Component = (
  <View
    entries={[
      {
        id: 1,
        hoursWorked: 8,
        notes: "Positive row",
        date: new Date("2020-06-22"),
      },
      {
        id: 1,
        hoursWorked: 4,
        notes: "Negative row",
        date: new Date("2020-06-22"),
      },
    ]}
    currentUser={{
      preferredWorkingHours: 6,
    }}
  />
);

test("render entries view", () => {
  let view = renderer.create(Component);
  let tree = view.toJSON();
  expect(tree).toMatchSnapshot();
});

test("entries with more hours worked than preferred hours is positive", () => {
  const { getByText } = render(Component);
  expect(
    getByText("Positive row").parentElement.classList.contains("positive")
  ).toBeTruthy();
});

test("entries with less hours worked than preferred hours is negative", () => {
  const { getByText } = render(Component);
  expect(
    getByText("Negative row").parentElement.classList.contains("negative")
  ).toBeTruthy();
});
