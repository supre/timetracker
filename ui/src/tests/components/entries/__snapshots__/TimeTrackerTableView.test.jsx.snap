// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render entries view 1`] = `
<table
  className="ui bottom attached table"
>
  <thead
    className=""
  >
    <tr
      className=""
    >
      <th
        className="three wide"
      >
        Date
      </th>
      <th
        className="three wide"
      >
        Hours Worked
      </th>
      <th
        className="eight wide"
      >
        Notes
      </th>
      <th
        className="two wide"
      >
        Action
      </th>
    </tr>
  </thead>
  <tbody
    className=""
  >
    <tr
      className="positive"
    >
      <td
        className=""
      >
        Mon Jun 22 2020
      </td>
      <td
        className=""
      >
        8
      </td>
      <td
        className=""
      >
        Positive row
      </td>
      <td
        className=""
      >
        <button
          className="ui basic circular icon button"
          onClick={[Function]}
        >
          <i
            aria-hidden="true"
            className="edit icon"
            onClick={[Function]}
          />
        </button>
        <button
          className="ui basic circular icon button"
          onClick={[Function]}
        >
          <i
            aria-hidden="true"
            className="delete icon"
            onClick={[Function]}
          />
        </button>
      </td>
    </tr>
    <tr
      className="negative"
    >
      <td
        className=""
      >
        Mon Jun 22 2020
      </td>
      <td
        className=""
      >
        4
      </td>
      <td
        className=""
      >
        Negative row
      </td>
      <td
        className=""
      >
        <button
          className="ui basic circular icon button"
          onClick={[Function]}
        >
          <i
            aria-hidden="true"
            className="edit icon"
            onClick={[Function]}
          />
        </button>
        <button
          className="ui basic circular icon button"
          onClick={[Function]}
        >
          <i
            aria-hidden="true"
            className="delete icon"
            onClick={[Function]}
          />
        </button>
      </td>
    </tr>
  </tbody>
</table>
`;
